tests/CMakeLists.txt

```diff
@@ -0,0 +1,172 @@
+## Copyright(c) 2017, Intel Corporation
+##
+## Redistribution  and  use  in source  and  binary  forms,  with  or  without
+## modification, are permitted provided that the following conditions are met:
+##
+## * Redistributions of  source code  must retain the  above copyright notice,
+##   this list of conditions and the following disclaimer.
+## * Redistributions in binary form must reproduce the above copyright notice,
+##   this list of conditions and the following disclaimer in the documentation
+##   and/or other materials provided with the distribution.
+## * Neither the name  of Intel Corporation  nor the names of its contributors
+##   may be used to  endorse or promote  products derived  from this  software
+##   without specific prior written permission.
+##
+## THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+## AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+## IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+## ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+## LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+## CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+## SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+## INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+## CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+## ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+## POSSIBILITY OF SUCH DAMAGE.
+
+cmake_minimum_required(VERSION 2.8.11)
+project(opae-tests)
+
+# Bootstrap gtest
+option(BUILD_GTEST "Enable gtest compilation" ON)
+mark_as_advanced(BUILD_GTEST)
+
+if (NOT BUILD_GTEST)
+  # Find gtest prevents compilation in travis CI
+  # Ubuntu gmock libraries are non-position relocatable
+  find_package(GTest)
+else()
+  message(STATUS "Trying to fetch gtest throught git")
+  find_package(Git REQUIRED)
+
+  # Enable ExternalProject CMake module
+  include(ExternalProject)
+
+  # Download and install GoogleTest
+  ExternalProject_Add(
+    gtest
+    GIT_REPOSITORY "https://github.com/google/googletest"
+    GIT_TAG "release-1.8.0"
+    UPDATE_COMMAND ""
+    PREFIX ${CMAKE_CURRENT_BINARY_DIR}/gtest
+    CMAKE_ARGS -DCMAKE_POSITION_INDEPENDENT_CODE=ON
+    # Disable install step
+    INSTALL_COMMAND "")
+  set (gtest_root "${CMAKE_CURRENT_BINARY_DIR}/gtest/src/gtest/googletest")
+  message(STATUS "gtest locatet at: ${gtest_root}")
+
+  # Create a libgtest target to be used as a dependency by test programs
+  add_library(libgtest IMPORTED STATIC GLOBAL)
+  add_dependencies(libgtest gtest)
+
+  # Get GTest source and binary directories from CMake project
+  ExternalProject_Get_Property(gtest source_dir binary_dir)
+
+  # Set libgtest properties
+  set_target_properties(libgtest PROPERTIES
+    "IMPORTED_LOCATION" "${binary_dir}/googlemock/gtest/libgtest.a"
```

Comment:
The extension will have to change from `.a` to `.so` if we build with `-DBUILD_SHARED_LIBS=ON`